### PR TITLE
Added individual suppression to spotbugs false positives

### PR DESCRIPTION
**Pull request by StefanSpieker**
Added individual suppression to spotbugs false positives of type `UNSAFE_HASH_EQUALS`

### Proposed changelog entries

- N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7124"></a>

### CODE REVIEW

**Review by NotMyFault — COMMENTED** [review]
What's the purpose behind this PR? I fail to see a reasoning, why four spotbugs silencers were added to the Java files in particular, rather than leaving them as-is.
Does that comply with a policy I am unaware of?

**Comment by StefanSpieker** [comment]
It is a follow up of https://github.com/jenkinsci/jenkins/pull/6015

**Review by basil — COMMENTED** [review]
Hi Stefan, first of all, thank you for picking up this effort to triage these static analysis warnings. We definitely appreciate the effort spent cleaning up this code, some of which is very old. In many cases the original author is not around to answer questions about it, so we have to do our best to infer the original intention. In any case this work is much appreciated.

That having been said, after taking a first pass at these justifications, I am not sure this PR is ready to merge as-is. I read the description of [`UNSAFE_HASH_EQUALS`](https://find-sec-bugs.github.io/bugs.htm#UNSAFE_HASH_EQUALS) on the Find Security Bugs web site and saw that the point of this check is to prevent an attacker from detecting the value of the secret hash due to the exposure of comparison timing. So in order to approve these justifications, I need to understand the reason why the exposure of comparison timing is not relevant in the methods in question. But the justifications you have provided indicate _what_ the method is doing (e.g., checking against true, generating a new token, or revocation), not _why_ the exposure of comparison timing is not relevant to that action.

At best, this is leaving the reasoning process as an exercise for the reader, and at least one reader (me) is having trouble deducing the reasoning (which could very well be due to my own deficiencies). At worst these justifications may be lacking in cogency. Assuming the best, I would ask that you clarify the reasoning more explicitly in the provided justifications. If this is asking too much, I would suggest not giving up, but merely trying to triage a different warning: some are a lot easier to reason about than others! But I should warn you that most of the ones that still remain are at least somewhat non-trivial, since if they were trivial I would have already fixed them by now. :smile:

File: core/src/main/java/jenkins/security/ApiTokenProperty.java
Comment (daniel-beck) [inline core/src/main/java/jenkins/security/ApiTokenProperty.java:178]:
```suggestion
    @SuppressFBWarnings(value = "UNSAFE_HASH_EQUALS", justification = "Used to prevent use of pre-2013 API tokens, then returning the API token value")
```

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
